### DISCUSSION TITLE
localhost doesn't work on WSL2

### DISCUSSION THREAD

**Discussion by mvjl000** [microsoft/WSL#6503] [General]
I'm using zsh WSL Ubuntu on windows 10, and when I try to run something on localhost (currently Node Express app) my browser can't find anything on http://localhost:8080 (port is correct). This same situation is in Create React App. No answer from http://localhost:3000, but 'On Your Network' http://...:3000 it works. However, when i switch to powershell, everything is as it should be.

# Environment

```none
Windows 10
Ubuntu 20.04
WSL2
```

# Steps to reproduce

npm start

#  Expected behavior

Response on http://localhost:8080

# Actual behavior

No response on http://localhost:8080, but it says:

> Error: listen EADDRINUSE: address already in use :::8080

**Comment by jung-youjin** [comment]
Hi! I've faced similar issue previously and turned out that port 8080 was already in use by other node. You can try `lsof -i :8080` or `lsof -c node` to view the list and kill whichever process that is in duplicate by `kill -9 "PID"`. Hope this helped!

**Comment by therealkenc** [comment]
> 'On Your Network' http://...:3000 it works

There is no repro here, but if `...` is your WSL2 ip address and it works, this is almost certainly a variation of #4851 (and/or #5298). The `EADDINUSE` on ipv6 will be because it is in use, especially if you are bouncing between Windows node and WSL node.

Use can use following to debug what is actually listening 

```
$ netstat -nlt
$ netstat.exe -a -n -p TCP
```
